keras_cv/callbacks/pycoco_callback.py

```diff
@@ -44,6 +44,12 @@ def __init__(self, validation_data, bounding_box_format, cache=True, **kwargs):
         self.bounding_box_format = bounding_box_format
         super().__init__(**kwargs)
 
+    def _num_detections(self, bounding_boxes):
+        boxes = bounding_boxes["boxes"]
+        if isinstance(boxes, tf.RaggedTensor):
+            return boxes.row_lengths(axis=1)
+        raise ValueError("Unimplemented")
```

Comment:
I think the consensus is the nms decoder always return "num_detections"?

Reply:
I coerce to ragged in line 67 by filtering -1s.

Reply:
I don't think theres a consensus here personally;  Is there any reason to not output ragged tensors from predict(), or at least let users decide which way?

I have supported num_detections as well now, but I don't think we should totally abandon outputting ragged boxes.